components/style/media_queries.rs

```diff
@@ -240,19 +242,43 @@ impl MediaQuery {
 /// media query list is only filled with the equivalent of "not all", see:
 ///
 /// https://drafts.csswg.org/mediaqueries/#error-handling
-pub fn parse_media_query_list(context: &ParserContext, input: &mut Parser) -> MediaList {
+pub fn parse_media_query_list<R>(
+    context: &ParserContext,
+    input: &mut Parser,
+    error_reporter: &R
+) -> MediaList
+where
+    R: ParseErrorReporter
+{
     if input.is_exhausted() {
         return MediaList::empty()
     }
 
     let mut media_queries = vec![];
     loop {
+        let start_position = input.position();
+        let start_location = input.current_source_location();
         match input.parse_until_before(Delimiter::Comma, |i| MediaQuery::parse(context, i)) {
             Ok(mq) => {
                 media_queries.push(mq);
             },
-            Err(..) => {
+            Err(err) => {
                 media_queries.push(MediaQuery::never_matching());
+                let error = match err {
+                    CssParseError::Custom(SelectorParseError::Custom(
+                            StyleParseError::MediaQueryExpectedFeatureName)) =>
+                        ContextualParseError::MediaQueryExpectedFeatureName(
+                            input.slice_from(start_position), err),
+                    CssParseError::Custom(SelectorParseError::Custom(
+                            StyleParseError::RangedExpressionWithNoValue)) =>
+                        ContextualParseError::MediaQueryNoMinMaxWithoutValue,
+                    CssParseError::Custom(SelectorParseError::Custom(
+                            StyleParseError::MediaQueryExpectedFeatureValue)) =>
+                        ContextualParseError::MediaQueryExpectedFeatureValue,
+                    _ => ContextualParseError::UnknownAtRule(input.slice_from(start_position), err),
```

Comment:
`UnknownAtRule` is not necessarily correct, is it? We can try to parse media query lists from a whole set of different places, like `@import` rules, stylesheet attributes...

Reply:
Ah, right. I guess something like a new `UnknownMediaQuery` will do then?

Reply:
Rather than matching on known errors here, we should simply have a `InvalidMediaRule(ParseError)` variant.